torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1075,7 +1075,17 @@ that adds a prefix to each key inserted to the store.
               "barrier",
               &::c10d::ProcessGroup::barrier,
               py::arg("opts") = ::c10d::BarrierOptions(),
-              py::call_guard<py::gil_scoped_release>());
+              py::call_guard<py::gil_scoped_release>())
+          .def(
+              "monitored_barrier",
+              []( const c10::intrusive_ptr<::c10d::ProcessGroup>& self, double timeoutSeconds) {
```

Comment:
Maybe we just use `timedelta` in the API so it automatically gets converted to std::chrono::... via pybind?

Reply:
For RPC framework, I think we ended up going with float for the timeout over timedelta, should we keep it consistent here?

Reply:
Looks like float is also needed to make this work with TorchScript. There is a separate TODO to make TorchBind work with std::chrono (https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/init.cpp#L1599), so we can probably scope this PR to only add python support and add torchscript support as a follow up.